webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -1868,10 +1868,6 @@ var userOverride = {
          * Where window.Applozic.ALSocket.events is the object we're referring to in the above scenario.
          */
         _this.subscribeToEvents = function (events, callback) {
-            if (!IS_SOCKET_CONNECTED) {
-                SUBSCRIBE_TO_EVENTS_BACKUP = events;
-                return;
-            }
```

Comment:
how was this causing issue?

Reply:
So the flow was : 

1. Initially the Socket is not connected. 
2. So in that if-condition, in the `SUBSCRIBE_TO_EVENTS_BACKUP` variable initially the custom events (added from widget installation script) gets stored.
3. In the meanwhile ZendeskChatService gets loaded and there different events gets subscribed .
4. ```  var events = {
            'onMessageSent': _this.handleUserMessage,
            'onMessageReceived': _this.handleBotMessage,
        };
        Kommunicate.subscribeToEvents(events);```
5. And now that `SUBSCRIBE_TO_EVENTS_BACKUP` will store the Zendesk Events only. 
6. And until this point socket gets connected and widget will only work with the lastly subscribed events. In our case that is Zendesk events.

This is how it was getting overriden.

Reply:
@raj5036 
Suppose the socket connects after some time, example 10 sec, then will the zendesk and custom events trigger?

Reply:
@gargshantnu I and Abhishek looked over these code changes and tested Chat-widget with this, it looks fine. We can move ahead with this.

cc: @Prochnost55

Reply:
Yes, the code was woking fine for normal case. 
The issue is for the time when there is delay when connecting to socket.